### ISSUE TITLE
read evidence analysis

### ISSUE THREAD

**Issue by timodonnell** [hammerlab/guacamole#287]
It could be useful to have an analysis that takes in a collection of BAM files and variants (specified in a set of VCF files) and writes out a csv with, for each BAM file, locus, the number of reads with each allele. This would be after applying some configurable read filters.

Currently, I have a Python [script](https://github.com/hammerlab/ovarian-cancer/blob/master/projects/pt189/read_evidence_plot.py) for generating this data for pt189 (something like 23 BAMs and ~1k variants). I ran it on all pt189 datasets like this:

```
time python read_evidence_plot.py \
    datasets.json \
    --evidence-summary-out out_full.csv \
    --extra-loci-offsets -1 \
    --sefara-set box_path_prefix=~/Box\ Sync/ \
```

(the `--extra-loci-offsets` argument tells it to, besides querying each locus with a variant, also query the locus before the variant, so we can plot non-variant loci near each variant).

It took 87 minutes to generate this 143k line output file:

```
(venv-2.7)[odonnt02@hammerlab-dev3 ~/sinai/git/ovarian-cancer/projects/pt189]$ head /hpc/users/odonnt02/sinai/git/ovarian-cancer/projects/pt189/out_full.csv
sample,contig,start,end,allele,count
bam_normal_exome_illumina,2,91673898,91673899,A,19
bam_normal_exome_illumina,2,91673897,91673898,T,19
bam_normal_exome_illumina,2,92315419,92315420,A,2
bam_normal_exome_illumina,2,92315419,92315420,C,4
bam_normal_exome_illumina,2,92315419,92315420,T,2629
bam_normal_exome_illumina,2,92315419,92315420,G,72
bam_normal_exome_illumina,2,92315418,92315419,A,1
bam_normal_exome_illumina,2,92315418,92315419,C,2
bam_normal_exome_illumina,2,92315418,92315419,T,92
(venv-2.7)[odonnt02@hammerlab-dev3 ~/sinai/git/ovarian-cancer/projects/pt189]$
```

I'd be interested if a Spark analysis for this could be significantly faster than 87 minutes.

**Comment by arahuja** [comment]
I'd be up for taking this on.

One question I have is why the out put is `bamfile`, `pos`, `counts` as opposed to each header being a BAM file?  I understand its pretty each to pivot it, but if this 100k+ variants, and you easily want to grab all the data for a single variant, having it pre-aggregated by position may be useful?

**Comment by timodonnell** [comment]
The alleles at a locus may be different for different BAM files, and there
may be any number of alleles at a locus (e.g. could have many different
indel sequences). So having one line per allele was the most
straightforward. If you see a better way to do it though, I can definitely
adapt my plotting script to use it, so anything works for me.

**Comment by arahuja** [comment]
Closed by #288
